bulk_insert.go

```diff
@@ -32,7 +32,48 @@ func BulkInsert(db *gorm.DB, objects []interface{}, chunkSize int, excludeColumn
 	return nil
 }
 
+// BulkInsertWithAssigningIDs executes the query to insert multiple records at once.
+// it will scan 'returning id' to [returningId] after every insert.
+// it's necessary to set insert_option="returning id" in *gorm.DB
+//
+// [returningId] must be a *[]uint(for integer) or *[]string(for uuid)
+//
+// [objects] must be a slice of struct.
+//
+// [chunkSize] is a number of variables embedded in query. To prevent the error which occurs embedding a large number of variables at once
+// and exceeds the limit of prepared statement. Larger size normally leads to better performance, in most cases 2000 to 3000 is reasonable.
+//
+// [excludeColumns] is column names to exclude from insert.
+func BulkInsertWithAssigningIDs(db *gorm.DB, returningId interface{}, objects []interface{}, chunkSize int, excludeColumns ...string) error {
```

Comment:
I think "returning" is a bit confusing for the caller since "returning" comes from DB side, just defining `ids interface{}` is clear for the caller.

Reply:
```
// it will scan the result of `returning id` or `returning *` to [returnedValue] after every insert.
// it's necessary to set "gorm:insert_option"="returning id" in *gorm.DB
// [returnedValue] slice of primary_key or model, must be a *[]uint(for integer), *[]string(for uuid), *[]struct(for `returning *`)
```
It can be more general in some case.  we can scan what we inserted using `returning *` back to the value, so i renamed it as `returnedValue` (i write more example in README). what do u think of it?